toolkit/tools/grapher/grapher.go

```diff
@@ -178,11 +178,23 @@ func addSingleDependency(g *pkggraph.PkgGraph, packageNode *pkggraph.PkgNode, de
 		}
 	}()
 	if newEdge.To() == newEdge.From() {
-		logger.Log.Warnf("Package %+v requires itself!", packageNode)
-	} else {
-		g.SetEdge(newEdge)
+		logger.Log.Debugf("Package %+v requires itself!", packageNode)
+		return nil
+	}
+
+	// Avoid creating runtime dependencies from an RPM to a different provide from the same RPM as the dependency will always be met on RPM installation.
+	// Creating these edges may cause non-problematic cycles that can significantly increase memory usage and runtime during cycle resolution.
+	// If there are enough of these cycles it can exhaust the system's memory when resolving them.
+	// - Only test run nodes as if a build node has a reflexive cycle then it cannot be built without a boostrap version.
```

Comment:
> - Only test run nodes as if a build node has a reflexive cycle then it cannot be built without a boostrap version. [](start = 3, length = 115)

Nit: this sentence blue-screen my brain. I'm not sure what it's about but this may due to the fact that I don't have detailed knowledge of this graph. Also: I think you've missed a "t" in "boostrap".

Reply:
Will clarify comment, it is definitely confusing